game/src/camera/camera_plugin.rs

```diff
@@ -0,0 +1,77 @@
+use avian2d::prelude::*;
+use bevy::{input::mouse::MouseWheel, prelude::*};
+
+pub struct CameraPlugin;
+
+//Component that marks our Camera's current target
+#[derive(Component)]
+pub struct CameraTarget;
+
+//Component that marks our main camera
+#[derive(Component)]
+pub struct MainCamera;
+
+impl Plugin for CameraPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(Startup, setup_main_camera);
+        app.add_systems(
+            PostUpdate,
+            update_camera
+                .after(PhysicsSet::Sync)
+                .before(TransformSystem::TransformPropagate),
+        );
+        app.add_systems(Update, zoom_camera);
+    }
+}
+
+//Sets up a 2d camera and attaches the MainCamera marker to it
+fn setup_main_camera(mut commands: Commands) {
+    commands.spawn((Camera2dBundle::default(), MainCamera));
+}
+
+//Update camera function will have our main camera follow any entity with the CameraTarget tag
+fn update_camera(
+    mut camera_query: Query<&mut Transform, (With<MainCamera>, Without<CameraTarget>)>,
+    target_query: Query<&Transform, (With<CameraTarget>, Without<MainCamera>)>,
+) {
+    let Ok(mut camera) = camera_query.get_single_mut() else {
+        error!("could not execute query for single MainCamera component");
+        return;
+    };
+
+    let Ok(target) = target_query.get_single() else {
+        error!("Could not execute query for single CameraTarget component");
+        return;
+    };
+
+    let Vec3 { x, y, .. } = target.translation;
+
+    let direction = Vec3::new(x, y, camera.translation.z);
+
+    camera.translation = direction;
+}
```

Comment:
After testing, the error message on line 43 keeps printing. The issue is that we if there is no `CameraTarget`, or more than one, then this message gets printed. 

I think at the minimum, we should only get an error whenever there is more than one camera target, and no errors if there is none. We'll figure out the rest later, but in the meantime you can fix this first.